lib/app-common.ts

```diff
@@ -115,22 +112,37 @@ export interface GlobalTunnelNgConfig {
  * default exclusion patterns are added for all private IPv4 address ranges.
  */
 async function setupGlobalHttpProxy(settings: CliSettings) {
-	const semver = await import('semver');
-	if (semver.lt(process.version, '10.16.0')) {
-		setupGlobalTunnelNgProxy(settings);
-	} else {
-		// use global-agent instead of global-tunnel-ng
-		await setupGlobalAgentProxy(settings);
+	// `global-tunnel-ng` accepts lowercase variables with higher precedence
+	// than uppercase variables, but `global-agent` does not accept lowercase.
+	// Set uppercase versions for backwards compatibility.
+	const { env } = process
+	if (env.http_proxy) {
```

Comment:
I moved this code to shared so that we can rely on just the single form of `HTTP_PROXY` for both proxy setups and also have to check less vars in the `if` below